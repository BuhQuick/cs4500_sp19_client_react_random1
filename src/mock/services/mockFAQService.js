var faqData = require('../data/faq.mock.json')

global.fetch = jest.fn()
  .mockImplementation(url => {
    if(url.includes("/faqs/102")) {
      return new Promise((resolve,reject) => {
        resolve({json:function() {
          return {
            "id":102,
            "question": "What is my second question?",
            "title": "Q2"
          }
        }});
      });
    }
    else if(url.includes("/faqs") {
      return new Promise(resolve => resolve(faqsData))

    }
  });
