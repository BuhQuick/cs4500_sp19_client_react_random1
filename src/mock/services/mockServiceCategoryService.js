var srvCatData = require('../data/service_categories.mock.json')

global.fetch = jest.fn()
    .mockImplementation((url, config) => {
        if(!config) {
            if(url.indexOf('api/categories/0') != -1) { 
                return srvCatData[0]
            } else if(url.indexOf('api/categories/1') != -1) {
                return srvCatData[1]
            } else if(url.indexOf('api/categories') != -1) {
                return new Promise((resolve, reject) => {
                    resolve({ json: function() {
                        return srvCatData
                    }})})
            }
        }});
