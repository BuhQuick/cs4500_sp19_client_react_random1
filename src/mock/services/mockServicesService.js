var srvData = require('../data/service.mock.json')

let findAllServices = () => new Promise(resolve => resolve(srvData))

export default findAllServices
