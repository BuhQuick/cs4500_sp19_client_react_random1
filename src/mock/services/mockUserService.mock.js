import users from '../MockData/Users.mock'

global.fetch = jest.fn().mockImplementation((url, config) => {
    if (!config) {
        if (url.indexOf('api/users/1') != -1) {
            return users[0];
        } else if (url.indexOf('api/users/10') != -1) {
            return users[1];
        } else if (url.indexOf('api/users/11') != -1) {
            return users[2];
        } else if (url.indexOf('api/users/12') != -1) {
            return users[3];
        } else if (url.indexOf('api/users') != -1) {
            return new Promise((resolve, reject) => {
                resolve({
                    json: function () {
                        return users;
                    }
                })
            })
        }
    } else if (config.method === 'put') {
         if (url.indexOf('api/users/1') != -1) {
             users[0] = JSON.parse(config.body);
             return users[0];
         } else if (url.indexOf('api/users/10') != -1) {
             users[1] = JSON.parse(config.body);
             return users[1];
         } else if (url.indexOf('api/users/11') != -1) {
             users[2] = JSON.parse(config.body);
             return users[2];
         }else if (url.indexOf('api/users/12') != -1) {
             users[3] = JSON.parse(config.body);
             return users[3];
         }
    } else if (config.method === 'delete') {
          if (url.indexOf('api/users/1') != -1) {
              return users[0];
          } else if (url.indexOf('api/users/10') != -1) {
              return users[1];
          } else if (url.indexOf('api/users/11') != -1) {
              return users[2];
          } else if (url.indexOf('api/users/12') != -1) {
              return users[3];
          }
    } else if (config.method === 'post') {
        let user = JSON.parse(config.body);
        user.id = (new Date()).getTime();
        users.push(user);
        return new Promise((resolve, reject) => {
            resolve({
                json: function () {
                    return users;
                }
            })
        })
    }
});