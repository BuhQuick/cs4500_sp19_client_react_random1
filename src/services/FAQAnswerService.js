import MIDDLE_TIER_URL from '../Env.js'
export default class FAQAnswerService {
    static instance = null;
    static getInstance() {
        if(FAQAnswerService.instance === null) {
            FAQAnswerService.instance = new FAQAnswerService()
        }
        return this.instance
    }
    findFAQAnswerById = id =>
        fetch(MIDDLE_TIER_URL + "/api/faq-answers/" + id)
            .then(response => response.json())
    findAllFAQAnswers = () =>
        fetch(MIDDLE_TIER_URL + "/api/faq-answers")
            .then(response => response.json())
}
