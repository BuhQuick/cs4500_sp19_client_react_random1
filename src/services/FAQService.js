import MIDDLE_TIER_URL from '../Env.js'
export default class FAQService {
    static instance = null;
    static getInstance() {
        if(FAQService.instance === null) {
            FAQService.instance = new FAQService()
        }
        return this.instance
    }
    findFAQById = id =>
        fetch(MIDDLE_TIER_URL + "/api/faqs/" + id)
            .then(response => response.json())
    findAllFAQs = () =>
        fetch(MIDDLE_TIER_URL + "/api/faqs")
            .then(response => response.json())
    filterFAQs = filterParams =>
        fetch(MIDDLE_TIER_URL + "/api/faqs/filter?title=" + filterParams.title + "&question=" + filterParams.question)
        .then(response => response.json())
    addFAQ = (newFAQ) =>
        fetch(MIDDLE_TIER_URL + "/api/faqs/create?title=" + newFAQ.title + "&question=" + newFAQ.question)
            .then(response => response.json());
};
