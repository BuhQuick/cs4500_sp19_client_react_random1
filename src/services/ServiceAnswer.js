import MIDDLE_TIER_URL from '../Env.js'
export default class ServiceAnswer {
    static instance = null;
    static getInstance() {
        if(ServiceAnswer.instance === null) {
            ServiceAnswer.instance = new ServiceAnswer()
        }
        return this.instance
    }
    findServiceAnswerById = serviceAnswerId =>
        fetch(MIDDLE_TIER_URL + "/api/admin/service-answers/" + serviceAnswerId)
            .then(response => response.json())
    findAllServiceAnswers = () =>
        fetch(MIDDLE_TIER_URL + "/api/admin/service-answers")
            .then(response => response.json())
}
