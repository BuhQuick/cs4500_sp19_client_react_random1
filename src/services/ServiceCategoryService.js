import MIDDLE_TIER_URL from '../Env.js'

export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if(ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }
    findServiceCategoryById = categoryId =>
        fetch(MIDDLE_TIER_URL + "/api/categories/" + categoryId)
            .then(response => response.json())
    findAllServiceCategories = () =>
        fetch(MIDDLE_TIER_URL + "/api/categories")
            .then(response => response.json())
}
