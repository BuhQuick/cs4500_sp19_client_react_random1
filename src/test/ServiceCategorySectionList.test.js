import React from 'react'
import ServiceCategorySectionList from '../components/serviceNavigator/ServiceCategorySectionList'
import renderer from 'react-test-renderer'
const srvCatData = require('../mock/data/service_categories.mock.json')
import ServiceCategoryService from '../services/serviceCategoryService'
const serviceCategoryService = ServiceCategoryService.getInstance()
import '../mock/services/mockServiceCategoryService'

test('service category section list renders correctly', () => {
    const testRenderer = renderer.create(
            <ServiceCategorySectionList
                serviceCategories={srvCatData}/>)

        let tree = testRenderer.toJSON()
        expect(tree).toMatchSnapshot()

        const testInstance = testRenderer.root

        const allCats = testInstance.findByProps({className: 'list-group no-border'})
        const svcSecs = testInstance.findAllByProps({className: 'list-group-item no-border'})

        expect(allCats).toBeDefined()
        expect(svcSecs.length).toBe(4)
})

