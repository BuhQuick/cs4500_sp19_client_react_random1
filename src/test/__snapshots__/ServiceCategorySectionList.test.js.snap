// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service category section list renders correctly 1`] = `
<ul
  className="list-group no-border"
>
  <li
    className="list-group-item no-border"
  >
    <div>
      <a
        id={3003}
      />
      <h2>
        House Cleaning
      </h2>
      <div>
        <div
          className="row"
        >
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Handymen
            </a>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li
    className="list-group-item no-border"
  >
    <div>
      <a
        id={3013}
      />
      <h2>
        Pet Services
      </h2>
      <div>
        <div
          className="row"
        >
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Pet Service 1
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Pet Sitters
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Dog Trainers
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Cat Tamers
            </a>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li
    className="list-group-item no-border"
  >
    <div>
      <a
        id={3023}
      />
      <h2>
        Tutoring & Training
      </h2>
      <div>
        <div
          className="row"
        >
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Brain Trainers
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Math Tutors
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Fitness Trainers
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Baseball Coashing
            </a>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li
    className="list-group-item no-border"
  >
    <div>
      <a
        id={3033}
      />
      <h2>
        Misc.
      </h2>
      <div>
        <div
          className="row"
        >
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Hypnosis
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Elemental Training
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Piano Lessons
            </a>
          </div>
          <div
            className="col-6 list-group-item no-border"
          >
            <a
              href="#"
            >
              Misc4
            </a>
          </div>
        </div>
      </div>
    </div>
  </li>
</ul>
`;
